test/inline_asm_clobbers.cl

```diff
@@ -0,0 +1,94 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown -x cl -cl-std=CL2.0 -O0 -emit-llvm-bc %s -o %t.bc
+// RUN: llvm-spirv -spirv-ext=+SPV_INTEL_inline_assembly %t.bc -o %t.spv
+// RUN: llvm-spirv %t.spv -to-text -o %t.spt
+// RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+// RUN: llvm-spirv -r %t.spv -o %t.bc
+// RUN: llvm-dis < %t.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+// Excerpt from opencl-c-base.h
+typedef __SIZE_TYPE__ size_t;
+
+// Excerpt from opencl-c.h to speed up compilation.
+#define __ovld __attribute__((overloadable))
+#define __cnfn __attribute__((const))
+size_t __ovld __cnfn get_global_id(unsigned int dimindx);
+
+// CHECK-SPIRV: {{[0-9]+}} Capability AsmINTEL
+// CHECK-SPIRV: {{[0-9]+}} Extension "SPV_INTEL_inline_assembly"
+// CHECK-SPIRV: {{[0-9]+}} AsmTargetINTEL
+
+// XCHECK-LLVM: [[STRUCTYPE:%[a-z0-9]+]] = type { i32, i32 }
+
+// CHECK-LLVM-LABEL: define spir_kernel void @mem_clobber
+// CHECK-SPIRV: {{[0-9]+}} AsmINTEL {{[0-9]+}} {{[0-9]+}} {{[0-9]+}} {{[0-9]+}} """~{cc},~{memory}"
+// CHECK-LLVM: [[VALUE:%[0-9]+]] = load i32 addrspace(1)*, i32 addrspace(1)**
+// CHECK-LLVM-NEXT: getelementptr inbounds i32, i32 addrspace(1)* [[VALUE]], i64 0
+// CHECK-LLVM-NEXT: store i32 1, i32 addrspace(1)*
+// CHECK-LLVM-NEXT: call void asm sideeffect "", "~{cc},~{memory}"()
+// CHECK-LLVM-NEXT: load i32 addrspace(1)*, i32 addrspace(1)**
+
+kernel void mem_clobber(global int *x) {
```

Comment:
I'm not really familiar with asm, so, probably silly question, but this function is named `clobber` and there is a decoration named `ClobberINTEL` in the spec proposal, but it is not implemented in this PR - is it intentional?

Reply:
Now everything works without this extension, including clobbers in assembly. We are doubt that it required at all and may be spec will be revisited to remove it. This implementation intentionally leaves it behind.